src/v/storage/compacted_index_writer.h

```diff
@@ -73,7 +73,6 @@ class compacted_index_writer {
 
     virtual ss::future<> append(compacted_index::entry) = 0;
 
-    virtual ss::future<> truncate(model::offset) = 0;
```

Comment:
for the active segment, spill key index truncation doesn't happen, rather the entire index seems as though it is deleted?